### PR TITLE
Adding "OpenProject" as a new Custom Service.

### PR DESCRIPTION
**Pull request by Bernhard-Krop**
Hi there!

As the title says, I've added "OpenProject" as a new custom service. I tested it well and everything works fine. The details are described in the Git commit message and should be reviewed.

Kind regards,

Bernhard

### CODE REVIEW

File: src/js/remoteServices.js [hunk 1]
```diff
@@ -115,6 +115,36 @@ export default {
...
+      if (subject && subjectId) {
+        subject = subjectId + " " + subject;
+      }
+
+      return subject || subjectId;
+    },
+    projectId: (document) => {
+      // ":project" in URL can be project name or OP internal project ID. Therefore, it cannot be used.
+      return (document.querySelector(".-project-context a") || document.querySelector("#projects-menu"))?.textContent?.trim() || "";
+    },
```
Comment (manubo) [inline src/js/remoteServices.js:143]: The `projectId` should return the MOCO project number. The project number should be added to the OpenProject project name in square brackets, e.g. `Demo project [P123456]`, where P123456 is the project number in MOCO, and be extracted from there. You can use the `projectRegex` for that, see the other examples.

Comment (manubo) [reply src/js/remoteServices.js:143]: I you would like to find a project by label, you might add an additional property `projectLabel` to the service. Square brackets in the project name should be removed, i.e. `Demo project [P1234566]` should become projectLabel `Demo project`

**Comment by manubo** [comment]
Thanks for your PR @Bernhard-Krop.
Could you please check my comments?

File: src/js/remoteServices.js [hunk 2]
```diff
@@ -115,6 +115,36 @@ export default {
...
+    allowHostOverride: true,
```
[the remark this answers is not in the document]
Comment (Bernhard-Krop) [reply src/js/remoteServices.js:144]: Yes, OpenProject can be installed on a self-hosted server and we actually do this.

File: src/js/components/Options.js [hunk 3]
```diff
@@ -152,7 +152,7 @@ class Options extends Component {
                 <div className="input-group">
                   <span
                     className="input-group-addon input-group-addon--left"
-                    style={{ display: "inline-block", width: "70px", textAlign: "left" }}
+                    style={{ display: "inline-block", width: "100px", textAlign: "left" }}
```
[the remark this answers is not in the document]
Comment (Bernhard-Krop) [reply src/js/components/Options.js:155]: The minimum width of the text "Openproject" is 82px. due to the font and the last letter being a "t", the width can be reduced to 78px. The name fields do have a padding of 12px each left and right. The sum is 102px. I rounded this down to 100px.
We could reduce the left and right padding to also reduce the element's width. A padding of 5px would reduce the width to 88px, which could be rounded up to 90px. A padding of 1px would reduce the width to exactly 80px. What do you think?

File: src/js/remoteServices.js [hunk 4]
```diff
@@ -115,6 +115,36 @@ export default {
     allowHostOverride: false,
   },
 
+  openproject: {
+    name: "openproject",
+    host: "",
```
[the remark this answers is not in the document]
Comment (Bernhard-Krop) [reply src/js/remoteServices.js:120]: Thank you, I was looking for the IP address but couldn't find it. I'll add it.

File: src/js/remoteServices.js (hunk 1 above)
Comment (Bernhard-Krop) [reply src/js/remoteServices.js:143]: Nice idea. I will add a new property `projectLabel`.

**Comment by Bernhard-Krop** [comment]
I've answered to each of your comments, @manubo. I left one question and am working on the others.

**Comment by Bernhard-Krop** [comment]
@manubo , I've implemented the new property `proejctLabel` and added remote service host for "openproject". I would like you to review my changes and mark your comments as resolved, if they are resolved. As soon as you answered my question regarding the service name width, I'll apply it in a new commit.

File: src/js/components/Options.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (manubo) [reply src/js/components/Options.js:155]: This is how Options look with 100px:
Screenshot_27_04_23__14_45

And this is how it looks with 80px:
Screenshot_27_04_23__14_46

80px is fine.

**Comment by manubo** [comment]
Thanks @Bernhard-Krop. I have answered regarding the width. We can keep it simple and just increase it to 80px. I have attached screenshots here and it looks fine with 80px, do you agree?

**Comment by Bernhard-Krop** [comment]
@manubo , it seems like there is a bug in one of the two browsers (see my message in your comment). This should be investigated, but I won't have much time starting tomorrow for 1-2 weeks. Either we wait until I have time to fix it, or we keep it 100px now and fix it later.

**Comment by manubo** [comment]
@Bernhard-Krop This PR has been released in Version 2.3.0. It's already available for Firefox, Chrome will follow.

File: src/js/components/Options.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bernhard-Krop) [reply src/js/components/Options.js:155]: I've found the solution. Chrome is using "content-box" as default for CSS attribute "box-sizing", while Firefox is using "border-box" as default. I set it to "content-box" and reduced the width back to 80px.

I'll provide the fix in a new PR, together with some other fixes / wishes I got from colleagues, due to the first release with OpenProject. I'll also mark this conversation as resolved. If there is any further issue, I would like to discuss it in a conversation in the new PR.

**Pull request merged**